app/domain/people/duplicate_checker.rb

```diff
@@ -0,0 +1,28 @@
+# encoding: utf-8
+
+class People::DuplicateChecker
+  def check
+    Person.find_each do |person|
+      duplicate = person_duplicate_finder.find({ first_name: person.first_name,
+                                                 last_name: person.last_name,
+                                                 company_name: person.company_name,
+                                                 zip_code: person.zip_code,
+                                                 birthday: person.birthday })
+      
+      next if person == duplicate || doublette_already_exists?(person, duplicate)
+
+      PersonDuplicate.create!(person_1: person, person_2: duplicate)
+    end
+  end
+
+  private
+
+  def person_duplicate_finder
+    @person_duplicate_finder ||= Import::PersonDuplicateFinder.new
+  end
+
+  def doublette_already_exists?(person_1, person_2)
```

Comment:
you could collect all person_ids where there's duplicate entries for initially. 
of course, you have to update it when you're creating new duplicate_person entries.
this way you would reduce the numbers of sql queries generated significantly. (1 instead of 1000s)